app/controllers/groupings_controller.rb

```diff
@@ -34,14 +51,27 @@ def update
     redirect_to edit_grouping_path
   end
 
+  def destroy
+    if @grouping.destroy
+      redirect_to groupings_path
+    else
+      flash[:error] = @grouping.errors.full_messages
+      redirect_to edit_grouping_path
+    end
+  end
+
   private
 
   def load_grouping
     @grouping = Grouping.find(params.require(:id))
   end
 
+  def new_grouping_params
+    params.require(:grouping).permit(:parent_grouping_id, :name)
+  end
+
   def grouping_params
-    params[:grouping].permit(:background)
+    params[:grouping].permit(:background, :parent_grouping_id, :name)
```

Comment:
While you're here, we should require `grouping` so that we don't run into a `KeyError` when `:grouping` doesnt exist in the params hash